### ISSUE TITLE
Compilation error

### ISSUE THREAD

**Issue by flurin** [elixir-cldr/cldr_messages#18]
While trying to update to the latest version (1.0.1) we're getting compile errors:

```
==> ex_cldr_messages
Compiling 11 files (.ex)

== Compilation error in file lib/cldr/messages/parser/parser.ex ==
** (UndefinedFunctionError) function Cldr.Number.PluralRule.known_plural_types/0 is undefined (module Cldr.Number.PluralRule is not available)
    Cldr.Number.PluralRule.known_plural_types()
    lib/cldr/messages/parser/parser.ex:13: (module)
```

It looks like it's dependin on `ex_cldr` which we don't use. But even if we add `ex_cldr` to the dependencies of our application it doesn't work.

We did not have any problems with version 0.14.1

This is running on:

```
Erlang/OTP 26 [erts-14.1.1] [source] [64-bit] [smp:10:10] [ds:10:10:10] [async-threads:1] [jit]

Elixir 1.15.7 (compiled with Erlang/OTP 26)
```

**Comment by kipcole9** [comment]
Thanks for the report @flurin, and my apologies for the inconvenience. This issue arises from [this commit](https://github.com/elixir-cldr/cldr_messages/commit/f462ffed7626daeacd7c95db0677ddfe76e12631) which removes [ex_cldr_numbers](https://github.com/elixir-cldr/cldr_numbers) as a dependency.

It's definitely a bug. I assume for now this hasn't been raised by other users because they are configuring either `ex_cldr_dates_times` to do date time formatting or `ex_money` to do money formatting. Because configuring either of those will also configure `ex_cldr_numbers` and therefore not see this bug.

I have two ways that I can fix this bug:

1. Make `ex_cldr_numbers` fully optional with conditional compilation.  
2. Restore `ex_cldr_numbers` as a required dependency.

Do you have an opinion?  Do your messages call for formatting numbers?

**Comment by flurin** [comment]
Well looking at the code it `Cldr.Number.PluralRule.known_plural_types/0` is required during compile time making it impossible to compile `Cldr.Message.Parser` without it. So I think it should be restored to be a required dependency, right?

**Comment by kipcole9** [comment]
> Well looking at the code it Cldr.Number.PluralRule.known_plural_types/0 is required during compile time making it impossible to compile Cldr.Message.Parser without it. So I think it should be restored to be a required dependency, right?

I can always extract the constants returned by `Cldr.Number.PluralRule.known_plural_types/0` directly into the code so it's not a deal breaker. But it's not ideal either.

So I agree, restoring `ex_cldr_numbers` as a direct dependency makes more sense. I will do that now and publish an updated version within the hour.

**Comment by flurin** [comment]
Wow, thanks @kipcole9 that's much appreciated!

**Comment by kipcole9** [comment]
I've published [ex_cldr_messages version 1.0.2](https://hex.pm/packages/ex_cldr_messages/1.0.2) with the following changelog entry:

### Bug Fixes

* Restore `ex_cldr_numbers` as a required dependency. Thanks to @flurin for the report. Closes #18.

Thank again for the report and sorry for the overly aggressive dependency pruning.
